internal/services/appservice/linux_function_app_resource.go

```diff
@@ -1210,6 +1230,29 @@ func (r LinuxFunctionAppResource) CustomizeDiff() sdk.ResourceFunc {
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
 			client := metadata.Client.AppService.ServicePlanClient
 			rd := metadata.ResourceDiff
+			if rd.HasChange("vnet_image_pull_enabled") {
+				planId := rd.Get("service_plan_id")
+				// the plan id is known after apply during the initial creation
+				if planId.(string) == "" {
+					return nil
+				}
+				_, newValue := rd.GetChange("vnet_image_pull_enabled")
+				servicePlanId, err := commonids.ParseAppServicePlanID(planId.(string))
+				if err != nil {
+					return fmt.Errorf("reading service plan id %+v", err)
+				}
+
+				asp, err := client.Get(ctx, *servicePlanId)
+				if err != nil {
+					return fmt.Errorf("could not read Service Plan %s: %+v", servicePlanId, err)
```

Comment:
Simplify this error message like we do elsewhere
```suggestion
					return fmt.Errorf("retrieving %s: %+v", servicePlanId, err)
```